### PR TITLE
add windows ssse3,sse4_1,sse4_2 detection for non avx path

### PR DESCRIPTION
**Pull request by ajax16384**
This  PR allows to detect >sse3 for non-avx processors by using only windows methods. Which brings identical detection windows caps as of linux or macos.

### CODE REVIEW

**Comment by ajax16384** [comment]
BTW this detection may superseed WESTMERE arch detect https://github.com/google/cpu_features/blob/8eb944f55d768a3c0250a1e80bb94de7b2560971/src/impl_x86_windows.c#L48
so it might be removed at all (and   _WIN32_WINNT compile time dependency seems fragile)

**Comment by toor1245** [comment]
> BTW this detection may superseed WESTMERE arch detect
> 
> https://github.com/google/cpu_features/blob/8eb944f55d768a3c0250a1e80bb94de7b2560971/src/impl_x86_windows.c#L48
> 
> 
> so it might be removed at all (and _WIN32_WINNT compile time dependency seems fragile)

ah, my bad, when I was working on this problem I only referred to the Microsoft docs. You are right, there is no point in _WIN32_WINNT, it can be removed. Thanks for the PR 👍

File: test/cpuinfo_x86_test.cc [hunk 1]
```diff
@@ -20,6 +20,20 @@
...
+#endif
+
+#if !defined(PF_SSE4_1_INSTRUCTIONS_AVAILABLE)
+#define PF_SSE4_1_INSTRUCTIONS_AVAILABLE            37
+#endif
+
+#if !defined(PF_SSE4_2_INSTRUCTIONS_AVAILABLE)
+#define PF_SSE4_2_INSTRUCTIONS_AVAILABLE            38
+#endif
+
```
Comment (toor1245) [inline test/cpuinfo_x86_test.cc:36]: should we move this code to a common place to avoid duplication?

File: src/impl_x86_windows.c [hunk 2]
```diff
@@ -43,6 +56,15 @@ static void DetectFeaturesFromOs(X86Info* info, X86Features* features) {
...
+      GetWindowsIsProcessorFeaturePresent(PF_SSSE3_INSTRUCTIONS_AVAILABLE);
+  features->sse4_1 =
+      GetWindowsIsProcessorFeaturePresent(PF_SSE4_1_INSTRUCTIONS_AVAILABLE);
+  features->sse4_2 =
+      GetWindowsIsProcessorFeaturePresent(PF_SSE4_2_INSTRUCTIONS_AVAILABLE);
+
+// do not bother checking PF_AVX*
+// cause AVX enabled processor will have XCR0 be exposed and this function will be skipped at all
 
 // https://github.com/google/cpu_features/issues/200
```
Comment (toor1245) [inline src/impl_x86_windows.c:69]: this code can be removed as the new functionality replaces detection properly

File: test/cpuinfo_x86_test.cc (hunk 1 above)
Comment (toor1245) [reply test/cpuinfo_x86_test.cc:36]: no objections, @gchatelet what do you think?

Comment (ajax16384) [reply test/cpuinfo_x86_test.cc]: ok

File: include/internal/windows_utils.h [hunk 3]
```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 Google LLC
```
Comment (toor1245) [inline include/internal/windows_utils.h:1]: 2022?

File: test/cpuinfo_x86_test.cc [hunk 4]
```diff
@@ -1173,15 +1176,13 @@ TEST_F(CpuidX86Test, WIN_INTEL_WESTMERE_EX) {
...
-  EXPECT_FALSE(info.features.sse4_2);
-#else
+  EXPECT_TRUE(info.features.sse);
+  EXPECT_TRUE(info.features.sse2);
+  EXPECT_TRUE(info.features.sse3);
   EXPECT_TRUE(info.features.ssse3);
   EXPECT_TRUE(info.features.sse4_1);
   EXPECT_TRUE(info.features.sse4_2);
-#endif
+
```
Comment (toor1245) [inline test/cpuinfo_x86_test.cc:1185]: minor comment, extra space

Comment (ajax16384) [reply test/cpuinfo_x86_test.cc:1185]: ok

**Review by toor1245 — APPROVED** [review]

**Comment by gchatelet** [comment]
Thx @ajax16384 and @toor1245 !

**Comment by toor1245** [comment]
Created PR to update `IsProcessorFeaturePresent` docs https://github.com/MicrosoftDocs/sdk-api/pull/1251

**Pull request merged**